### DISCUSSION TITLE
Throwing an error when passing a SPACESHIP_CONNECT_API_KEY_FILEPATH that does not resolve

### DISCUSSION THREAD

**Discussion by philippeauriach** [fastlane/fastlane#18221] [Ideas]
Hello 🔑 ,

I just lost a lot of time on a new project because the secret `SPACESHIP_CONNECT_API_KEY_FILEPATH` was set but with a typo, which ended in a file path that does not exists.

When `SPACESHIP_CONNECT_API_KEY_FILEPATH` is set correctly, it is picked up automatically by fastlane to auth and injected in the context for other actions such as match and pilot to use.
But when this env variable is wrongly set, it just does not load the key without informing the user, and then match or pilot ask for the apple id and password in the "old way".

On the other hand, if one use the specific action, then it throws an error as the file is not found:
```
    app_store_connect_api_key(
      key_id: ENV["SPACESHIP_CONNECT_API_KEY_ID"],
      issuer_id: ENV["SPACESHIP_CONNECT_API_ISSUER_ID"],
      key_filepath: ENV["SPACESHIP_CONNECT_API_KEY_FILEPATH"]
    )
```
see https://github.com/fastlane/fastlane/blob/0d1aa50045d57975d8b9e5d5f1f489d82ee0f437/fastlane/lib/fastlane/actions/app_store_connect_api_key.rb#L61 

I would like to add an error (or at least a warning) if the user passes a non empty but non existent `SPACESHIP_CONNECT_API_KEY_FILEPATH`. It means in my opinion that the user made a mistake, and he should be informed of that.

The problem is, I don't have any idea where to add this check in the fastlane code, as I cannot figure where the AppStoreConnect token is created from the env variable. Any idea would be greatly appreciated.

**Comment by scotttball** [comment]
@philippeauriach I ran into this problem and this was the solution that worked for me.

```ruby
    app_store_connect_api_key(
      key_id: ENV["SPACESHIP_CONNECT_API_KEY_ID"],
      issuer_id: ENV["SPACESHIP_CONNECT_API_ISSUER_ID"],
      key_filepath: File.join(Dir.pwd, ENV["SPACESHIP_CONNECT_API_KEY_FILEPATH"]),
      in_house: false, # optional but may be required if using match/sigh
    )
```
